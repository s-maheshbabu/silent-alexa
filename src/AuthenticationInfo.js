import { hasIn } from "immutable";
import util from "util";

export default class AuthenticationInfo {
  constructor(authResponse) {
    if (arguments.length == 1 && this.validateAuthResponse(authResponse)) {
      var _access_token = authResponse.access_token;
      var _expires_in = authResponse.expires_in;
    }
    this.getAccessToken = function() {
      return _access_token;
    };
  }

  isValid() {
    return this.getAccessToken() ? true : false;
  }

  validateAuthResponse(authResponse) {
    if (
      hasIn(authResponse, ["access_token"]) &&
      hasIn(authResponse, ["expires_in"]) &&
      authResponse.access_token &&
      authResponse.expires_in
    ) {
      return true;
    } else {
      console.log(
        "Encountered an error on login: " +
          util.inspect(authResponse, { showHidden: true, depth: null })
      );
      return false;
    }
  }
}
