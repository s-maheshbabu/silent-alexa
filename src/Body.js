import React, { Component } from "react";
import MuiThemeProvider from "material-ui/styles/MuiThemeProvider";
import ChatWindow from "./ChatWindow";
import RightPanel from "./RightPanel";

export default class Body extends Component {
  render() {
    return [
      <MuiThemeProvider key="muiThemeProvider">
        <ChatWindow />
      </MuiThemeProvider>,

      <RightPanel key="rightPanel" />
    ];
  }
}
