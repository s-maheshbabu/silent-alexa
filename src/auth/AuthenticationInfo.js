import IllegalArgumentError from "errors/IllegalArgumentError";
import Cookies from "js-cookie";
import util from "util";

/**
 * Wrapper class to hold access_token and expires_in returned in LoginWithAmazon
 * authorization response.
 * Instances are created only if access_token and expires_in are defined. The token
 * is stored in a cookie called 'amazon_Login_accessToken' because that is the
 * cookie that LoginWithAmazon will look for.
 * https://developer.amazon.com/docs/login-with-amazon/javascript-sdk-reference.html#setUseCookie
 */
// TODO: Rename this to AuthenticationManager now that the class has been refactored to not
// actually hold the authentication information.
class AuthenticationInfo {
  persist(lwaResponse) {
    const numberOfSecondsInADay = 86400;

    if (lwaResponse && lwaResponse.access_token && lwaResponse.expires_in) {
      Cookies.set(AMAZON_LOGIN_COOKIE, lwaResponse.access_token, {
        expires: lwaResponse.expires_in / numberOfSecondsInADay,
        secure: false // TODO: Change localhost to also use https and then change this to true.
      });
    } else {
      const serializedLWAResponse = util.inspect(lwaResponse, {
        showHidden: true,
        depth: null
      });
      throw new IllegalArgumentError(
        `LoginWithAmazon Authorization response is undefined or 
          doesnt have access_token/expires_in.
          lwaResponse: ${serializedLWAResponse}`
      );
    }
  }

  /**
   * @returns true is authentication info is present and false otherwise.
   * This does not guarantee that the tokens haven't already expired.
   * Callers should be able to handle the case of tokens being
   * invalid when they try to use them.
   */
  isPresent = () => {
    return this.getAccessToken() !== undefined;
  };

  /**
   * @returns access_token to access Alexa API
   */
  getAccessToken = () => {
    return undefined;
  };

  /**
   * @returns deleted the cookie that stores the authentication info
   */
  clear = () => {
    Cookies.remove(AMAZON_LOGIN_COOKIE);
  };
}

// Declare a singleton
const instance = new AuthenticationInfo();
Object.freeze(instance);

export default instance;
export const AMAZON_LOGIN_COOKIE = "amazon_Login_accessToken";
