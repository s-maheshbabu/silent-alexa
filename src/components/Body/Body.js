import React from "react";
import AuthenticationInfo from "AuthenticationInfo";
import MuiThemeProvider from "material-ui/styles/MuiThemeProvider";
import ChatWindow from "ChatWindow/ChatWindow";
import RightPanel from "RightPanel/RightPanel";
import WelcomeScreen from "WelcomeScreen/WelcomeScreen";

export default function Body() {
  if (AuthenticationInfo.isPresent()) {
    return [
      <MuiThemeProvider key="muiThemeProvider">
        <ChatWindow />
      </MuiThemeProvider>,

      <RightPanel key="rightPanel" />
    ];
  }

  return <WelcomeScreen />;
}
