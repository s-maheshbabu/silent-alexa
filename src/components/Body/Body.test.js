import React from "react";
import { shallow, mount } from "enzyme";
import Body from "./Body";
import AuthenticationInfo from "AuthenticationInfo";

jest.mock("AuthenticationInfo");

beforeEach(() => {
  jest.resetAllMocks();
});

it("renders Body when AuthenticationInfo is present", () => {
  const isPresentMock = jest.spyOn(AuthenticationInfo, "isPresent");
  isPresentMock.mockImplementation(() => true);

  const wrapper = shallow(<Body />);
  expect(wrapper).toMatchSnapshot();

  wrapper.unmount();
});

it("renders Body when AuthenticationInfo is not present", () => {
  const isPresentMock = jest.spyOn(AuthenticationInfo, "isPresent");
  isPresentMock.mockImplementation(() => false);

  const wrapper = shallow(<Body />);
  expect(wrapper).toMatchSnapshot();

  wrapper.unmount();
});
