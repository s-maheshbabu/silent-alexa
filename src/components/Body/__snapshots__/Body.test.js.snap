// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Body when AuthenticationInfo is not present 1`] = `<WelcomeScreen />`;

exports[`renders Body when AuthenticationInfo is present 1`] = `
Array [
  <MuiThemeProvider
    key="muiThemeProvider"
/>,
  <RightPanel
    key="rightPanel"
/>,
]
`;
